bcdi/simulation/supportMaker.py

```diff
@@ -0,0 +1,414 @@
+
+import numpy as np
+import h5py as h5
+import sys
+
+helptext = """
+The concept is to be able to build a support from a set of defined planes these
+planes can be positioned based on physical size (nm) if known or eventually made
+into a fancy tool with a 3D view of the support versus the data so you can match
+fringes. It may also be interesting to consider how to make 3D supports from
+other characterisation methods (SEM etc...).
+"""
+
+####################################################################
+
+
+def AddPolyCen(array, center, planes):
+    """
+
+    Make the polygon.
+
+    :param : array = input array
+    :param : center = origin of polygon
+    :param : planes = array of planes
+
+    """
+    dims = array.shape
+
+    griddims = []
+    for d in dims:
+        griddims.append(slice(0, d))
+    grid = np.ogrid[griddims]
+
+    for plane in planes:
+        sum1 = np.zeros(dims)
+        sum2 = 0
+        for d in range(len(dims)):
+            sum1 += plane[d] * (grid[d] - center[d])
+            sum2 += plane[d] ** 2
+        array += (sum1 <= sum2) * 1
+
+    return ((array >= len(planes)) * 1).astype(array.dtype)
+
+
+####################################################################
+def MakePoly(dims, planes):
+    """
+
+    Make a polygon.
+
+    :param : dims = dimensions of array in pixels
+    :param : planes = array of planes
+
+    """
+    return make_poly(dims, planes)
+
+
+####################################################################
+def make_poly(dims, planes):
+    """
+
+    Make a polygon.
+
+    :param : dims = dimensions of array in pixels
+    :param : planes = array of planes
+
+    """
+    cen = []
+    array = np.zeros(dims)
+    for dim in dims:
+        cen.append(dim / 2)
+    return AddPolyCen(array, cen, planes)
+
+
+####################################################################
+def MakePolyCen(dims, center, planes):
+    """
+
+    Make the polygon.
+
+    :param : array = input array
+    :param : center = origin of polygon
+    :param : planes = array of planes
+
+    """
+    array = np.zeros(dims)
+    return AddPolyCen(array, center, planes)
+
+
+class supportMaker:
+    """
+
+    A masking class for support creation.
+
+    :param : rawdata
+    :param : x-ray wavelength
+    :param : detector_distance - sample to detector (m)
+    :param : detector_pixel_size - 2D (m)
+    :param : ang_step (degrees)
+    :param : braggAng (degrees)
+    :param : planes = array of planes [x,y,z]
+    :param : planesDist = array of plane distance to origin (m)
+    :param : voxel_size = set the voxel size to some arbitrary size,
+             np.array([x,y,z]) (m)
+
+    """
+
+    def __init__(
+        self,
+        rawdata,
+        wavelength=None,
+        detector_distance=None,
+        detector_pixel_size=None,
+        ang_step=None,
+        braggAng=None,
+        planes=None,
+        planesDist=None,
+        voxel_size=None,
+    ):
+        """Init supportMaker."""
```

Comment:
It's a detail: the docstring of __init__ is not rendered in the doc, therefore you can skip it.

```suggestion
```